plugin/evm/limitorders/limit_order_tx_processor.go

```diff
@@ -102,95 +100,40 @@ func NewLimitOrderTxProcessor(txPool *core.TxPool, memoryDb LimitOrderDatabase,
-
-	nonce := lotp.txPool.Nonce(common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC")) // admin address
-
-	data, err := lotp.orderBookABI.Pack("settleFunding")
-	if err != nil {
-		log.Error("abi.Pack failed", "err", err)
-		return err
-	}
-	key, err := crypto.HexToECDSA("56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027") // admin private key
-	if err != nil {
-		log.Error("HexToECDSA failed", "err", err)
-		return err
-	}
-	settleFundingTx := types.NewTransaction(nonce, lotp.orderBookContractAddress, big.NewInt(0), 5000000, big.NewInt(80000000000), data)
-	signer := types.NewLondonSigner(big.NewInt(321123))
-	signedTx, err := types.SignTx(settleFundingTx, signer, key)
-	if err != nil {
-		log.Error("types.SignTx failed", "err", err)
-		return err
-	}
-	err = lotp.txPool.AddLocal(signedTx)
-	if err != nil {
-		log.Error("types.SignTx failed", "err", err)
-		return err
-	}
-	return nil
-
+	return lotp.executeOrderBookLocalTx("settleFunding")
 }
 
 func (lotp *limitOrderTxProcessor) ExecuteMatchedOrdersTx(incomingOrder LimitOrder, matchedOrder LimitOrder, fillAmount *big.Int) error {
-	//randomly selecting private key to get different validator profile on different nodes
-	rand.Seed(time.Now().UnixNano())
-	var privateKey, userAddress string
-	if rand.Intn(10000)%2 == 0 {
-		privateKey = privateKey1
-		userAddress = userAddress1
-	} else {
-		privateKey = privateKey2
-		userAddress = userAddress2
-	}
-
-	nonce := lotp.txPool.Nonce(common.HexToAddress(userAddress)) // admin address
-	ammID := big.NewInt(1)
 	orders := make([]Order, 2)
 	orders[0], orders[1] = getOrderFromRawOrder(incomingOrder.RawOrder), getOrderFromRawOrder(matchedOrder.RawOrder)
+
 	signatures := make([][]byte, 2)
 	signatures[0] = incomingOrder.Signature
 	signatures[1] = matchedOrder.Signature
 
-	data, err := lotp.orderBookABI.Pack("executeMatchedOrders", ammID, orders, signatures, fillAmount)
+	return lotp.executeOrderBookLocalTx("executeMatchedOrders", orders, signatures, fillAmount)
+}
+
+func (lotp *limitOrderTxProcessor) executeOrderBookLocalTx(method string, args ...interface{}) error {
+	nonce := lotp.txPool.Nonce(common.HexToAddress(userAddress1)) // admin address
+
+	data, err := lotp.orderBookABI.Pack(method, args...)
 	if err != nil {
 		log.Error("abi.Pack failed", "err", err)
 		return err
 	}
-	key, err := crypto.HexToECDSA(privateKey) // admin private key
+	key, err := crypto.HexToECDSA(privateKey1) // admin private key
 	if err != nil {
 		log.Error("HexToECDSA failed", "err", err)
 		return err
 	}
-	executeMatchedOrdersTx := types.NewTransaction(nonce, lotp.orderBookContractAddress, big.NewInt(0), 5000000, big.NewInt(80000000000), data)
-	signer := types.NewLondonSigner(big.NewInt(321123))
-	signedTx, err := types.SignTx(executeMatchedOrdersTx, signer, key)
+	tx := types.NewTransaction(nonce, lotp.orderBookContractAddress, big.NewInt(0), 0, big.NewInt(80000000000), data)
```

Comment:
gasPrice should be 0, not gasLimit. gaslimit should be 5mil

Reply:
yes, missed it earlier. Changed now